test/interactions/keyInteractionTests.ts

```diff
@@ -139,5 +139,136 @@ describe("Interactions", () => {
 
         svg.remove();
     });
+
+    it("triggers appropriate callback for key released", () => {
+      var svg = TestMethods.generateSVG(400, 400);
+      var component = new Plottable.Component();
+      component.renderTo(svg);
+
+      var keyInteraction = new Plottable.Interactions.Key();
+
+      var aCode = 65; // "a" key
+      var bCode = 66; // "b" key
+
+      var aCallbackCalled = false;
+      var aCallback = () => aCallbackCalled = true;
+      var bCallbackCalled = false;
+      var bCallback = () => bCallbackCalled = true;
+
+      keyInteraction.onKeyRelease(aCode, aCallback);
+      keyInteraction.onKeyRelease(bCode, bCallback);
+      keyInteraction.attachTo(component);
+
+      var $target = $(component.background().node());
+
+      TestMethods.triggerFakeMouseEvent("mouseover", component.background(), 100, 100);
+      $target.simulate("keyup", { keyCode: aCode });
+      assert.isFalse(aCallbackCalled, "callback for \"a\" was not called when \"a\" key was released without being pressed");
+
+      $target.simulate("keydown", { keyCode: aCode });
+      $target.simulate("keyup", { keyCode: aCode });
+      assert.isTrue(aCallbackCalled, "callback for \"a\" was called when \"a\" key was released after being pressed");
+
+      aCallbackCalled = false;
+      $target.simulate("keydown", { keyCode: bCode });
+      $target.simulate("keyup", { keyCode: bCode });
+      assert.isFalse(aCallbackCalled, "callback for \"a\" was not called when \"b\" key was released");
+      assert.isTrue(bCallbackCalled, "callback for \"b\" was called when \"b\" key was released");
+
+      bCallbackCalled = false;
+      $target.simulate("keydown", { keyCode: aCode });
+      TestMethods.triggerFakeMouseEvent("mouseout", component.background(), -100, -100);
+      $target.simulate("keyup", { keyCode: aCode });
+      assert.isTrue(aCallbackCalled,
+        "callback for \"a\" was called when \"a\" key was released outside of component after being pressed inside");
+
+      aCallbackCalled = false;
+      $target.simulate("keydown", { keyCode: aCode });
+      $target.simulate("keyup", { keyCode: aCode });
+      assert.isFalse(aCallbackCalled,
+        "callback for \"a\" was not called when \"a\" key was released after being pressed outside");
+
+      $target.simulate("keydown", { keyCode: aCode });
+      TestMethods.triggerFakeMouseEvent("mouseover", component.background(), 100, 100);
+      $target.simulate("keyup", { keyCode: aCode });
+      assert.isFalse(aCallbackCalled,
+        "callback for \"a\" was not called when \"a\" key was released inside of component after being pressed outside");
+
+      svg.remove();
+    });
```

Comment:
I see a huge test suite here that is decently not too readable.  Is it possible to split this test up so that the test suite is concise and more organized?  Using a mixture of `describe` blocks or so can accomplish this.